hatchery/hatchery.go

```diff
@@ -55,16 +57,65 @@ func paymodels(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	userName := getCurrentUserName(r)
-	payModel, err := getPayModelForUser(userName)
+
+	payModel, err := getCurrentPayModel(userName)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
 	if payModel == nil {
-		http.Error(w, err.Error(), http.StatusNotFound)
+		http.Error(w, "Current paymodel not set", http.StatusNotFound)
 		return
 	}
+	out, err := json.Marshal(payModel)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	out, err := json.Marshal(payModel)
+	fmt.Fprint(w, string(out))
+}
+
+func allpaymodels(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		http.Error(w, "Not Found", http.StatusNotFound)
+		return
+	}
+	userName := getCurrentUserName(r)
+
+	payModels, err := getPayModelsForUser(userName)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if payModels == nil {
+		http.Error(w, "No paymodel set", http.StatusNotFound)
```

Comment:
is this also a `NopaymodelsError`?

Reply:
I think it might be 🤔  I need to double check if the function is returning `NopaymodelsError` or not. Maybe we can change this accordingly.